### PR TITLE
[bazel][java] Remove TestNG dependency

### PR DESCRIPTION
**Pull request by joerg1985**
### Description
This PR will remove TestNG, there is no change for the jars generated by bazel. I had to add `log4j-core:2.20.0` otherwise 3.0.0-alpha1 would be picked.

### Motivation and Context
Selenium switched from TestNG to JUnit in 2012, it is time to remove it.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by titusfortner — APPROVED** [review]

File: java/maven_deps.bzl [hunk 1]
```diff
@@ -97,7 +97,7 @@ def selenium_java_deps():
             "org.mockito:mockito-core:4.11.0",
             "org.slf4j:slf4j-api:2.0.7",
             "org.slf4j:slf4j-jdk14:2.0.7",
-            "org.testng:testng:7.7.1",
+            "org.apache.logging.log4j:log4j-core:2.20.0",
```
Comment (diemol) [inline java/maven_deps.bzl:100]: Why do we need log4j? Are we using that transitively?

Comment (joerg1985) [reply java/maven_deps.bzl:100]: `com.github.spotbugs:spotbugs` does use it and so it is added transitively.

Comment (titusfortner) [reply java/maven_deps.bzl:100]: Is this required for our linting tests? Why do we have it as a runtime dependency?

Comment (diemol) [reply java/maven_deps.bzl:100]: We should also get rid of spotbugs and use something across the project.

Also, those are all the dependencies, when the packages are created then only the runtime are packed.

**Review by diemol — APPROVED** [review]
Thank you, @joerg1985!

**Pull request merged**